GsaGHTests/2_GooWrappers/GsaElement2dGooTests.cs

```diff
@@ -1,32 +1,108 @@
 ﻿using Grasshopper.Kernel.Types;
 
+using GsaAPI;
+
 using GsaGH.Components;
 using GsaGH.Parameters;
 
 using GsaGHTests.Components.Geometry;
 using GsaGHTests.Helpers;
+using GsaGHTests.Parameters;
+
+using Rhino.Geometry;
+using Rhino.Geometry.Morphs;
 
 using Xunit;
 
 namespace GsaGHTests.GooWrappers {
   [Collection("GrasshopperFixture collection")]
   public class GsaElement2dGooTests {
-    [Fact]
-    public void GsaElement2dGooDrawViewportMeshesAndWiresTest() {
-      var comp = (Section3dPreviewComponent)CreateElement2dTests.ComponentMother();
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void GsaElement2dGooDrawViewportMeshesAndWiresTest(bool isLoadPanel) {
+      var comp = (Section3dPreviewComponent)CreateElement2dTests.ComponentMother(isLoadPanel, isLoadPanel);
       comp.Preview3dSection = true;
       var output = (GsaElement2dGoo)ComponentTestHelper.GetOutput(comp);
       GH_OasysGeometryGooTests.DrawViewportMeshesAndWiresTest(output);
     }
 
+    [Theory]
+    [InlineData(true, true)]
+    [InlineData(true,false)]
+    [InlineData(false,true)]
+    [InlineData(false, false)]
+    public void GetGeometryTest(bool isLoadPanel, bool preview3dSection) {
+      var comp = (Section3dPreviewComponent)CreateElement2dTests.ComponentMother(isLoadPanel, isLoadPanel);
+      comp.Preview3dSection = preview3dSection;
+      var output = (GsaElement2dGoo)ComponentTestHelper.GetOutput(comp);
+      Assert.NotNull(output.GetGeometry());
+    }
+
     [Fact]
     public void GsaElement2dGooCastToMeshTest() {
       var comp = (Section3dPreviewComponent)CreateElement2dTests.ComponentMother();
       var output = (GsaElement2dGoo)ComponentTestHelper.GetOutput(comp);
-
       var mesh = new GH_Mesh();
       Assert.True(output.CastTo(ref mesh));
       Assert.True(mesh.IsValid);
     }
+
+    [Fact]
+    public void GsaElement2dGooCastToCurveTest() {
+      var comp = (Section3dPreviewComponent)CreateElement2dTests.ComponentMotherLoadPanel();
+      var output = (GsaElement2dGoo)ComponentTestHelper.GetOutput(comp);
+      var curve = new GH_Curve();
+      Assert.True(output.CastTo(ref curve));
+      Assert.True(curve.IsValid);
+    }
+
+    [Fact]
+    public void MorphMeshTest() {
```

Comment:
could we give a more descriptive name for these tests (Morph....) What are they meant to test?

for instance

`StretchingASection3dShouldOutputAValidMesh` 

Although I don't understand what we are testing, as the first part is always the same and the transformation we are changing (morph through stretch or translation) are Rhino functions, which we can assume they work.

Reply:
done